test/routes/authentication.js

```diff
@@ -203,29 +181,66 @@ describe('/login POST', function () {
                 error.response.status.should.equal(401)
             })
     })
-})
 
-describe('/logout', function () {
-    it('should sign out', function (done) {
-        const agent = chai.request.agent(app)
-        agent.post('/signup').send(goodGuyGreg)
-            .end(function (error, res) {
-                res.status.should.be.equal(200)
-                agent.get('/user').send()
-                    .end(function (error, res) {
+    it('should save session after login', function (done) {
+        // Create a new user:
+        User.create(goodGuyGreg)
+            .then(function () {
+
+                // Login the new user:
+                const agent = chai.request.agent(app)
+                agent.post('/login').send({
+                    email: goodGuyGreg.email,
+                    password: goodGuyGreg.password
+                })
+                    .end(function (err, res) {
                         res.status.should.be.equal(200)
-                        agent.post('/logout').send()
-                            .end(function () {
+                        expect(res).to.have.cookie('winder-session')
+
+                        // Check that the user is now logged-in:
+                        agent.get('/user').send()
+                            .end(function (err, res) {
                                 res.status.should.be.equal(200)
-                                agent.get('/user').send()
-                                    .end(function (error, res) {
-                                        res.status.should.be.equal(200)
-                                        expect(res.body.user).to.be.undefined
-                                        done()
-                                    })
+                                expect(res).to.be.json
+                                expect(res.body).to.deep.include({
+                                    user: {
+                                        firstName: goodGuyGreg.firstName,
+                                        lastName: goodGuyGreg.lastName,
+                                        email: goodGuyGreg.email,
+                                        isSuperuser: false
+                                    }
+                                })
+                                done()
                             })
                     })
-
             })
     })
+
+})
+describe('/logout', function () {
+
```

Comment:
🤣 

yea apparently testing erroneous flows asynchronically in Chai is a bit tricky, haven't found a tidy way to do it yet.